### PR TITLE
Two sum algo

### LINKED ISSUE #203: [FEAT]: Implement Two Sum Algorithm in Cairo
# Implement Two Sum Algorithm in Cairo

## 📘 Description
Implement the classic Two Sum algorithm in Cairo. This algorithm finds two numbers in an array that add up to a specific target value - a common interview question and practical utility for smart contract development.

## ✅ Acceptance Criteria

- Return the indices of the two numbers that add up to the target.
- Add comprehensive tests for various scenarios:
- Valid inputs with a solution
- Valid inputs with no solution
- Edge cases (empty array, single element)
- Include performance comparison between the two approaches.

## 🌎 References
https://book.cairo-lang.org/
https://leetcode.com/problems/two-sum/description/

## 📜 Additional Notes
- ⚠ Read our guidelines before applying.

### PR DESCRIPTION
**Pull request by ryzen-xp**
# 📝 Pull Request Title

## 🛠️ Issue
- Closes #203 

## 📖 Description
Implemented the classic Two Sum algorithm in Cairo. This algorithm finds two numbers in an array that add up to a specific target value - a common interview question and practical utility for smart contract development.

## 🖼️ Screenshots (if applicable)

Screenshot from 2025-03-22 14-43-27

## 📝 Additional Notes
- Any additional context, questions, or considerations for the reviewers:

### CODE REVIEW

**Review by coxmars — CHANGES_REQUESTED** [review]
Add a .tool-versions file please, and also add some unit tests there and we are good 🫡

File: examples/cairo/scripts/two_sum_algorithm/.tool-versions [hunk 1]
```diff
@@ -0,0 +1,3 @@
+scarb 2.9.2
+cairo 2.9.2
+sierra 1.6.0
```
Comment (coxmars) [inline examples/cairo/scripts/two_sum_algorithm/.tool-versions:3]: Add a new line at the end of this file

**Review by coxmars — CHANGES_REQUESTED** [review]

File: examples/cairo/scripts/two_sum_algorithm/tests/test_two_sum_algo.cairo [hunk 2]
```diff
@@ -0,0 +1,55 @@
+use core::array::ArrayTrait;
+use super::*;
+pub fn two_sum(nums: Array<u128>, target: u128) -> (u32, u32) {
```
Comment (coxmars) [inline examples/cairo/scripts/two_sum_algorithm/tests/test_two_sum_algo.cairo:3]: Why did you create the same code here? Try to import the function from two_sum_algorithm, once it is done I will merge, thanks 🫡

**Review by coxmars — CHANGES_REQUESTED** [review]

**Review by coxmars — APPROVED** [review]
@ryzen-xp thank you for your work, merged 🫡
